lib/blkback.ml

```diff
@@ -118,8 +111,8 @@ end
 let service_thread t stats =
 
   let grants_of_segments = List.map (fun seg -> {
-    Gnttab.domid = t.domid;
-    ref = Int32.to_int seg.Req.gref;
+    OS.Xen.Import.domid = t.domid;
+    ref = Gntref.of_int32 seg.Req.gref;
```

Comment:
I think it would be better to change `Blkproto`, replacing any `int32` there that is a grant ref with `Gntref.t`. Then we wouldn't need these conversions in the main code.

Reply:
(see https://github.com/talex5/mirage-block-xen/commit/d72825198783b68f178b486964d80dea8ed4706d)